api/src/test/java/org/openmrs/module/fhir2/providers/r3/ProcedureRequestFhirResourceProviderTest.java

```diff
@@ -79,4 +96,107 @@ public void getServiceRequestByWithWrongId_shouldThrowResourceNotFoundException(
 		assertThat(resourceProvider.getProcedureRequestById(idType).isResource(), is(true));
 		assertThat(resourceProvider.getProcedureRequestById(idType), nullValue());
 	}
+
+	
+	@Test
+	public void searchServiceRequests_shouldReturnMatchingServiceRequests() {
+		serviceRequest = new ServiceRequest();
+		serviceRequest.setId(SERVICE_REQUEST_UUID);
+		
+		when(serviceRequestService.searchForServiceRequests(any(), any(), any(), any(), any()))
+		        .thenReturn(new MockIBundleProvider<>(Collections.singletonList(serviceRequest), 10, 1));
+		
+		TokenAndListParam code = new TokenAndListParam();
+		TokenParam codingToken = new TokenParam();
+		codingToken.setValue("1000");
+		code.addAnd(codingToken);
+		
+		IBundleProvider results = resourceProvider.searchForProcedureRequests(null, null, code, null, null, null);
+		assertThat(results, notNullValue());
+		assertThat(results.getResources(1, 5), hasSize(equalTo(1)));
+		assertThat(results.getResources(1, 5).get(0), notNullValue());
+		assertThat(results.getResources(1, 5).get(0).fhirType(), equalTo("ServiceRequest"));
+		assertThat(results.getResources(1, 5).get(0).getIdElement().getIdPart(), equalTo(SERVICE_REQUEST_UUID));
+		
+	}
+	
+	@Test
+	public void searchServiceRequests_shouldReturnMatchingServiceRequestsWhenPatientParamIsSpecified() {
+		serviceRequest = new ServiceRequest();
+		serviceRequest.setId(SERVICE_REQUEST_UUID);
+		
+		when(serviceRequestService.searchForServiceRequests(any(), any(), any(), any(), any()))
+		        .thenReturn(new MockIBundleProvider<>(Collections.singletonList(serviceRequest), 10, 1));
+		
+		ReferenceAndListParam patientParam = new ReferenceAndListParam();
+		patientParam.addValue(new ReferenceOrListParam().add(new ReferenceParam().setChain(Patient.SP_NAME)));
+		
+		IBundleProvider results = resourceProvider.searchForProcedureRequests(patientParam, null, null, null, null, null);
+		assertThat(results, notNullValue());
+		assertThat(results.getResources(1, 5), hasSize(equalTo(1)));
+		assertThat(results.getResources(1, 5).get(0), notNullValue());
+		assertThat(results.getResources(1, 5).get(0).fhirType(), equalTo("ServiceRequest"));
```

Comment:
We can also have START_INDEX and END_INDEX as constants like has been done with other resources.